gui/src/components/StoryComponents/StoryDetails.tsx

```diff
@@ -134,9 +155,49 @@ export default function StoryDetails({
         setStoryOverview(data.story.overview);
         setStoryTestCases(data.story.test_cases);
         setStoryInstructions(data.story.instructions);
+
+        if (data.story.status === storyStatus.IN_REVIEW) {
```

Comment:
so are using the same code snippet for design and backend story, why dont you extract this piece of code and use it in both

Reply:
this is not yet resolved. The code snippets are very identical